utils/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -230,7 +230,10 @@ private fun ZipFile.unpack(targetDirectory: File) =
                 continue
             }
 
-            val target = targetDirectory.resolve(entry.name)
+            // Strip absolute path spec to ensure that the file ends up in [targetDirectory].
+            val relativeFilePath = entry.name.trimStart('/')
```

Comment:
What about absolute Windows paths, could those also appear in ZIP files? And shouldn't we maybe instead skip such files rather than making them relative? They might create a clash with real relative entries after all.

So, I'd instead propose to amend line 229 with ` || File(entry.name).isAbsolute`.

And we should probably do the same for 7-Zip and `InputStream` unpacking, too.

Reply:
Good point update the Windows one.

Regarding skipping, the disadvantage would be that some files would not be examined at all - but I'd be fine with that.
@mnonnenmacher you as well?

Reply:
I ended-up implementing all your suggestions.